File: algorithms/1_easy/M/01217_minimum_cost_to_move_chips_to_the_same_position/greedy.cpp
class Solution {
public:
    static int minCostToMoveChips(const vector<int>& position) {
        assert(std::ssize(position) <= 100);
        int cnt[2] = { 0 };

        for (const int p : position) {
            cnt[(p & 1)] += 1;
        }

        return std::min(cnt[0], cnt[1]);
    }
};

File: algorithms/1_easy/M/01217_minimum_cost_to_move_chips_to_the_same_position/greedy.rs
impl Solution {
    pub fn min_cost_to_move_chips(position: Vec<i32>) -> i32 {
        let mut cnt = [0_isize; 2];

        for p in position {
            cnt[(p & 1) as usize] += 1;
        }

        cnt[0].min(cnt[1]).try_into().unwrap()
    }
}
